tsfile/src/main/java/org/apache/iotdb/tsfile/read/query/executor/TsFileExecutor.java

```diff
@@ -54,6 +58,25 @@ public TsFileExecutor(IMetadataQuerier metadataQuerier, IChunkLoader chunkLoader
 
   @Override
   public QueryDataSet execute(QueryExpression queryExpression) throws IOException {
+    // bloom filter
+    BloomFilter bloomFilter = metadataQuerier.getWholeFileMetadata().getBloomFilter();
+    List<Path> filteredSeriesPath = new ArrayList<>();
+    if(bloomFilter != null) {
+      for (Path path : queryExpression.getSelectedSeries()) {
+        if (bloomFilter.contains(path.getFullPath())) {
+          filteredSeriesPath.add(path);
+        }
+      }
+      if (filteredSeriesPath.isEmpty()) {
+        // return an empty QueryDataSet
+        LOG.warn(
+            "This tsfile not contains all paths which specified in selected series, return empty dataset");
```

Comment:
`warn` may be too high. Querying old files with newly created time series may often result in this branch, I think `debug` is enough.